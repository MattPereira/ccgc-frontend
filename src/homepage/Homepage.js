import React, { useContext } from "react";
import UserContext from "../auth/UserContext";
import { Link } from "react-router-dom";

const Homepage = () => {
  const { currentUser } = useContext(UserContext);
  console.debug("Homepage");

  return (
    <div className="text-center">
      <h1 className="display-3">Contra Costa Golf Club</h1>
      {currentUser ? (
        <p>
          Welcome back, {currentUser.firstName} {currentUser.lastName}!
        </p>
      ) : (
        <>
          <span>Please </span>
          <Link to="/register">register</Link>
          <span> or </span> <Link to="/login">login</Link>
          <span> to input your round scores!</span>
        </>
      )}
      <div>
        <img
          src="https://i.ibb.co/72nKCSf/ccgc-Logo11.png"
          alt="ccgc-logo"
          className="img-fluid"
        />
      </div>
    </div>
  );
};

export default Homepage;
