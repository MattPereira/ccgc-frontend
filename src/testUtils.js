import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "happy-gilmore",
  firstName: "Happy",
  lastName: "Gilmore",
  email: "happy@gmail.com",
  isAdmin: true,
  rounds: [],
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
