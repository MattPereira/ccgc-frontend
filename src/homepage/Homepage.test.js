import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { UserProvider } from "../testUtils";
import Homepage from "./Homepage";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("initially displays loading spinner", async function () {
  render(
    <MemoryRouter initialEntries={["/courses/yocha-dehe"]}>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );

  const ccgcLogo = screen.getByAltText("ccgc-logo");
  expect(ccgcLogo).toBeInTheDocument();
});
