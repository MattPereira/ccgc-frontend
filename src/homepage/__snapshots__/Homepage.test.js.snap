// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <div>
      <img
        alt="ccgc-logo"
        class="img-fluid"
        src="https://i.ibb.co/72nKCSf/ccgc-Logo11.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
